### ISSUE TITLE
Docker task improvement

### ISSUE THREAD

**Issue by jcovoluntis** [microsoft/azure-pipelines-tasks#13166]
## Required Information

**Question, Bug, or Feature?**  
*Type*: Question

**Enter Task Name**: Docker

In order to execute our Selenium based automated tests I have to run Selenoid docker image. 
If I understood docker correctly, I will have to "run" the container at the 1st execution, but only "start" it at the next executions (because it has been created already by the "run" command).
Would it be possible that the docker task handle that "run or start logic" for us ? 

So basically a logic like that : 

```
if [[ $(docker ps -a --filter="name=$CONTAINER_NAME" --filter "status=exited" | grep -w "$CONTAINER_NAME") ]]; then
    docker start...
else
    docker run...
fi
```

**Comment by ds-ms** [comment]
You could configure this at the step level right? You could use expressions to skip or run that particular `docker run` task step?
Here's some documentation: https://docs.microsoft.com/en-us/azure/devops/pipelines/process/expressions?view=azure-devops

Something like:

```yaml
script: |
   if [[ $(docker ps -a --filter="name=$CONTAINER_NAME" --filter "status=exited" | grep -w "$CONTAINER_NAME") ]]; then
      echo '##vso[task.setvariable variable=docker_start]true'
   else
      echo '##vso[task.setvariable variable=docker_start]false'
   fi

- task: Docker@v2
  condition: eq(variables.docker_start, 'true')
  inputs:
      command: start

- task: Docker@v2
  condition: eq(variables.docker_start, 'false')
  inputs:
      command: run
```

PS: This is just a pseudo syntax

**Comment by ds-ms** [comment]
Closing stale issue. Feel free to reopen if you're still stuck
